src/views/common/Animation/AnimationControls/PlaybackPositionArrowButtons.tsx

```diff
@@ -1,37 +1,29 @@
 import { faBackward, faFastBackward, faFastForward, faForward } from '@fortawesome/free-solid-svg-icons'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import React, { useCallback, useMemo } from 'react'
-import { AnimationStateDispatcher } from '../AnimationStateReducer'
-
-const usePlaybackPositionArrowButtons = (dispatch: AnimationStateDispatcher<any>) => {
-    const handleArrowClick = useCallback((mode: 'end' | 'skip', backward?: boolean) => {
-        return (e: React.MouseEvent) => {
-            dispatch({
-                type: mode === 'end' ? 'TO_END' : 'SKIP',
-                backward: backward
-            })
-        }
-    }, [dispatch])
+import { useMemo } from 'react'
+import { PlaybackControlLogic } from './PlaybackControlButtonLogic'
 
+const usePlaybackPositionArrowButtons = (logic: PlaybackControlLogic) => {
+    const {returnToBeginningHandler, backSkipHandler, forwardSkipHandler, jumpToEndHandler} = logic
     const beginningButton = useMemo(() => 
-        <span onMouseDown={handleArrowClick('end', true)} title="Return to beginning">
+        <span onMouseDown={returnToBeginningHandler} title="Return to beginning">
             <FontAwesomeIcon icon={faFastBackward} />
-        </span>, [handleArrowClick])
+        </span>, [returnToBeginningHandler])
 
     const backSkipButton = useMemo(() => 
-        <span onMouseDown={handleArrowClick('skip', true)} title="Return to beginning">
+        <span onMouseDown={backSkipHandler} title="Return to beginning">
             <FontAwesomeIcon icon={faBackward} />
-        </span>, [handleArrowClick])
+        </span>, [backSkipHandler])
 
     const forwardSkipButton = useMemo(() => 
-        <span onMouseDown={handleArrowClick('skip')} title="Skip forward">
+        <span onMouseDown={forwardSkipHandler} title="Skip forward">
             <FontAwesomeIcon icon={faForward} />
-        </span>, [handleArrowClick])
+        </span>, [forwardSkipHandler])
 
     const endButton = useMemo(() => 
-        <span onMouseDown={handleArrowClick('end')} title="Skip to end">
+        <span onMouseDown={jumpToEndHandler} title="Skip to end">
             <FontAwesomeIcon icon={faFastForward} />
-        </span>, [handleArrowClick])
+        </span>, [jumpToEndHandler])
```

Comment:
Since these no longer rely on a shared callback in this file to define their logic, it might be better to break them up into separate files. Eh.